addons/web_editor/static/src/components/media_dialog/file_selector.js

```diff
@@ -232,11 +229,21 @@ export class FileSelector extends Component {
         return this.props.selectedMedia[this.props.id].filter(media => media.mediaType === 'attachment').map(({ id }) => id);
     }
 
+    get mediaDomain() {
+        const domain = [
+            ['res_model', '=', this.props.resModel],
+            ['res_id', 'in', [0, this.props.resId]],
+            ['hidden', '=', false],
+        ];
+        return domain;
+    }
+
     get attachmentsDomain() {
         const domain = [
             '&',
-            ['res_model', '=', this.props.resModel],
-            ['res_id', '=', this.props.resId || 0]
+            ['res_model', '=', 'web_editor.media'],
+            '|', '&', ['res_field', '=', 'binary_content'], ['type', '=', 'binary'],
+                ['type', '=', 'url'],
         ];
```

Comment:
Just wondering, is there no Javascript equivalent of the `from odoo.osv.expression import AND, OR` tools ?

Reply:
Maybe `import { Domain } from "@web/core/domain";`

Reply:
I don't understand when/how to use it, especially when all the fields are spelled out like here. TBH, it's a pain to write  and makes it less readable in my view. (The parameter of `Domain.and()` and `Domain.or()` must be an array, which itself contains full domains (so nested arrays), which are either composed by just one array or several arrays. And in the end I still need to cast it to a list anyway, for the logic of the other modules to work. Am I missing something? I must be missing something, it doesn't seems right.)

```js
get attachmentsDomain() {
    let domain = Domain.and([
        [['res_model', '=', 'web_editor.media']],
        Domain.or([
            [["res_field", "=", "binary_content"], ["type", "=", "binary"]],
            [['type', '=', 'url']],
        ]),
    ]);
    domain = Domain.or([[['public', '=', true]], domain]);
    domain = Domain.and([domain, [['name', 'ilike', this.state.needle]]]);
    return domain.toList();
}
```

Reply:
It's a bit less worse if you import `or` and `and`.
Not great indeed, but at least it's a way to get rid of the `push` and the `unshift`... and it makes the tree structure visible (at least for me - but I tend to see everything as trees, so I might be non-representative of the general audience for this code).
```js
return and([
    [['name', 'ilike', this.state.needle]],
    or([
        [['public', '=', true]],
        and([
            [['res_model', '=', 'web_editor.media']],
            or([
                [['res_field', '=', 'binary_content'], ['type', '=', 'binary']],
                [['type', '=', 'url']],
            ]),
        ]),
    ]),
]).toList();
```

Reply:
> It's a bit less worse if you import or and and.

You can't import a class' static method.
I think I'll leave it as it is because all the other uses of `attachmentsDomain` do `push` and `unshift`, so let's keep the same writing "convention" in all of them. If further reviewers agree with your suggestion, I'll change it.